src/producto/producto.controller.ts

```diff
@@ -0,0 +1,28 @@
+import { Controller, Post, Get, Res, HttpStatus, Body } from '@nestjs/common';
+import { Request, Response } from 'express';
+import { CreateProductoDTO } from './dto/producto.dto';
+import { Producto } from './schemas/producto.schema';
+import { ProductoService } from './producto.service';
+
+@Controller('producto')
+export class ProductoController {
+  constructor(private readonly productoService: ProductoService) {}
+
+  @Get()
+  async getProducto(): Promise<Producto[]> {
+    const productos = await this.productoService.findAll();
+
+    return productos;
+  }
+
+  @Post()
+  async create(@Res() res: Response, @Body() createProductoDTO: CreateProductoDTO) {
+    let producto: Producto;
+    try {
+      producto = await this.productoService.create(createProductoDTO);
+      res.status(HttpStatus.CREATED).json(producto);
```

Comment:
Creo que hay un decorador para cambiar el estatus sin tener que usar `res`. Déjame reviso y te confirmo.

Reply:
Nest tiene el decorador `HttpCode`  y una serie de [HttpExceptions](https://docs.nestjs.com/exception-filters#built-in-http-exceptions), usando esos dos podemos evitar utilizar `res`, te voy dejando más o menos como usarlo